examples/training/semantic_segmentation/pascal_voc/basic_segmentation_training.py

```diff
@@ -0,0 +1,194 @@
+    "Whether or not to use FP16 mixed precision for training.",
+)
+flags.DEFINE_boolean(
+    "use_ema",
+    True,
+    "Whether or not to use exponential moving average weight updating",
+)
+flags.DEFINE_boolean(
+    "include_rescaling",
+    True,
+    "Whether or not to include rescaling in the model",
+)
+flags.DEFINE_string(
+    "backbone_weights",
+    "imagenet",
+    "Path for the weights to be loaded into the backbone",
+)
+flags.DEFINE_string(
+    "backbone",
+    None,
+    "String denoting a supported backbone for the segmentation model",
+)
+flags.DEFINE_float(
+    "initial_learning_rate",
+    0.007,
+    "Initial learning rate which will reduce on plateau. This will be multiplied by the number of accelerators in use",
+)
+flags.DEFINE_string(
+    "model_kwargs",
+    "{}",
+    "Keyword argument dictionary to pass to the constructor of the model being trained",
+)
+
+flags.DEFINE_float(
+    "weight_decay",
+    5e-4,
+    "Weight decay parameter for the optimizer",
+)
+
+# An upper bound for number of epochs (this script uses EarlyStopping).
+flags.DEFINE_integer("epochs", 1000, "Epochs to train for")
+
+FLAGS = flags.FLAGS
+FLAGS(sys.argv)
+
+# Try to detect an available TPU. If none is present, default to MirroredStrategy
+try:
+    tpu = tf.distribute.cluster_resolver.TPUClusterResolver.connect()
+    strategy = tf.distribute.TPUStrategy(tpu)
+    if FLAGS.mixed_precision:
+        logging.info("Mixed precision training enabled")
+        tf.keras.mixed_precision.set_global_policy("mixed_bfloat16")
+except ValueError:
+    # MirroredStrategy is best for a single machine with one or multiple GPUs
+    strategy = tf.distribute.MirroredStrategy()
+    if FLAGS.mixed_precision:
+        logging.info("Mixed precision training enabled")
+        tf.keras.mixed_precision.set_global_policy("mixed_float16")
+print("Number of accelerators: ", strategy.num_replicas_in_sync)
+
+local_batch = FLAGS.batch_size
+global_batch = local_batch * strategy.num_replicas_in_sync
+base_lr = FLAGS.initial_learning_rate * strategy.num_replicas_in_sync
+
+train_ds = load(split="sbd_train", data_dir=None)
+eval_ds = load(split="sbd_eval", data_dir=None)
+
+
+def preprocess_image(img, cls_seg, augment=False):
+    img = tf.keras.layers.Resizing(512, 512, interpolation="nearest")(img)
+    cls_seg = tf.keras.layers.Resizing(512, 512, interpolation="nearest")(cls_seg)
+    cls_seg = tf.cast(cls_seg, tf.uint8)
+
+    inputs = {"images": img, "segmentation_masks": cls_seg}
+    if augment:
+        inputs = preprocessing.RandomFlip("horizontal")(inputs)
+        inputs = preprocessing.RandomColorDegeneration(factor=0.3)(inputs)
+        inputs = preprocessing.RandomGaussianBlur(kernel_size=3, factor=0.3)(inputs)
+        inputs = preprocessing.RandomRotation(factor=0.1, segmentation_classes=21)(
+            inputs
+        )
+
+    return inputs["images"], inputs["segmentation_masks"]
+
+
+def proc_train_fn(examples):
+    image = examples.pop("image")
```

Comment:
Maybe we should just pass `examples` here and pop in the preprocessing function instead?